Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/Allocator/Jobs/DriverFeeUpdates/DriverFee.hs

```diff
@@ -0,0 +1,119 @@
+{-
+ Copyright 2022-23, Juspay India Pvt Ltd
+
+ This program is free software: you can redistribute it and/or modify it under the terms of the GNU Affero General Public License
+
+ as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version. This program
+
+ is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+
+ or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more details. You should have received a copy of
+
+ the GNU Affero General Public License along with this program. If not, see <https://www.gnu.org/licenses/>.
+-}
+
+module SharedLogic.Allocator.Jobs.DriverFeeUpdates.DriverFee
+  ( sendPaymentReminderToDriver,
+    unsubscribeDriverForPaymentOverdue,
+  )
+where
+
+import Domain.Types.DriverFee
+import Domain.Types.Person
+import qualified Kernel.External.Notification.FCM.Types as FCM
+import Kernel.Prelude
+import qualified Kernel.Storage.Esqueleto as Esq
+import Kernel.Storage.Hedis (HedisFlow)
+import Kernel.Tools.Metrics.CoreMetrics
+import Kernel.Types.Error
+import Kernel.Types.Id (Id, cast)
+import Kernel.Utils.Common (EsqDBFlow, Log (withLogTag), MonadFlow, MonadTime (getCurrentTime), addUTCTime, logInfo, throwError)
+import Lib.Scheduler
+import SharedLogic.Allocator
+import Storage.CachedQueries.CacheConfig (HasCacheConfig)
+import Storage.Queries.DriverFee
+import Storage.Queries.DriverInformation (updateSubscription)
+import qualified Storage.Queries.Person as QPerson
+import qualified Tools.Notifications as Notify
+
+sendPaymentReminderToDriver ::
+  ( MonadFlow m,
+    HedisFlow m r,
+    CoreMetrics m,
+    HasCacheConfig r,
+    EsqDBFlow m r,
+    Esq.EsqDBReplicaFlow m r
+  ) =>
+  Job 'SendPaymentReminderToDriver ->
+  m ExecutionResult
+sendPaymentReminderToDriver Job {id, jobInfo} = withLogTag ("JobId-" <> id.getId) do
+  let jobData = jobInfo.jobData
+      startTime = jobData.startTime
+      endTime = jobData.endTime
+  (driverFees, driverFeeIds, feeZipDriver, _) <- calcDriverFeeAttr ONGOING startTime endTime
+  if not $ null driverFees
+    then do
+      for_ feeZipDriver $ \(driverFee, mbDriver) -> do
+        case mbDriver of
+          Nothing -> do
+            logInfo "Driver Not found. This should not be possible."
+            throwError (InternalError "Driver Not Found") -- Unreachable
+          Just driver -> do
+            let paymentTitle = "Payment Pending"
+                paymentMessage = "Payment of Rs. " <> show driverFee.totalAmount.getMoney <> " is pending. Pay by " <> show driverFee.payBy <> " to avoid being blocked."
+            Notify.sendNotificationToDriver driver.merchantId FCM.SHOW Nothing FCM.PAYMENT_PENDING paymentTitle paymentMessage driver.id driver.deviceToken
```

Comment:
can you catch this ? because if any one fails all next would not be sent any notification.